11_notify/willshen8/pkg/puppy/lostpuppy.go

```diff
@@ -0,0 +1,35 @@
+package puppy
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+)
+
+// Payload struct is the format that lost puppy server expects to see
+type Payload struct {
+	ID int `json:"id"`
+}
+
+// HandleLostPuppy processes the payload and determine the response based on id field
+func HandleLostPuppy(w http.ResponseWriter, r *http.Request) {
+	time.Sleep(2 * time.Second)
+	var jsonData Payload
+	if decodeErr := render.DecodeJSON(r.Body, &jsonData); decodeErr != nil {
```

Comment:
Just use `err` instead of `decodeErr`. Always use `err` for error variables unless you need to have multiple at once. As you get more experience with reading Go code, you expect to see `err` for errors and when you do, it is very obvious at a glance what is happening. When you see a different name, you look deeper to try to understand why it is different to usual and it breaks your reading flow when there is no reason for it.

Reply:
Got it!